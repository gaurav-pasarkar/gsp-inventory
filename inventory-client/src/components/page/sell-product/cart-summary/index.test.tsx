import {ProductToAdd} from "../index";
import matchSnapshot from "../../../../test-utils/match-snapshot";
import CartSummary from "./index";
import {render} from "@testing-library/react";

describe('Cart summary', () => {

  const productsInCart = new Map<string, ProductToAdd>([
    ['Product 1',
      {
        quantityToAdd: 1,
        product: {
          productName: 'Product 1',
          sellingPrice: 10,
          costPrice: 9,
          quantity: 10
        }
      }
    ],
    ['Product 2',
      {
        quantityToAdd: 2,
        product: {
          productName: 'Product 2',
          sellingPrice: 20,
          costPrice: 15,
          quantity: 10
        }
      }
    ]
  ]);

  it('should render', () => {
    matchSnapshot(<CartSummary products={productsInCart}/>)
  });

  it('should show correct total', () => {
    const {getByText} = render(<CartSummary products={productsInCart}/>);
    expect(getByText(/Total: â‚¹ 50 \/-/)).toBeInTheDocument();
  });

  it('should show correct number of products on checkout', () => {
    const {getByText} = render(<CartSummary products={productsInCart}/>);
    expect(getByText(/Checkout \(2\)/)).toBeInTheDocument();
  })
});