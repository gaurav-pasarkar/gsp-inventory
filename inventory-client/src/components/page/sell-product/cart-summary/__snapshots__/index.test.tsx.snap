// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart summary should render 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <span
      className="MuiTypography-root MuiTypography-body1"
    >
      <div
        className="MuiBox-root MuiBox-root-2"
      >
        Total: â‚¹ 
        50
         /-
      </div>
    </span>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Checkout (
        2
        )
      </span>
    </button>
  </div>
</div>
`;
